typewise_alert.test.py

```diff
@@ -3,8 +3,17 @@
 
 
 class TypewiseTest(unittest.TestCase):
-  def test_infers_breach_as_per_limits(self):
-    self.assertTrue(typewise_alert.infer_breach(20, 50, 100) == 'TOO_LOW')
+  def test_breach_type(self):
+    self.assertTrue(typewise_alert.classify_temperature_breach('PASSIVE_COOLING', 34) == "NORMAL")
+    self.assertTrue(typewise_alert.classify_temperature_breach('HIGH_ACTIVE_COOLING',90) == "TOO_HIGH")
+    self.assertTrue(typewise_alert.classify_temperature_breach('PASSIVE_COOLING', -10) == "TOO_LOW")
+    self.assertTrue(typewise_alert.classify_temperature_breach('Passive_heat_sinks', 69)== "Wrong Input")
+    self.assertTrue(typewise_alert.classify_temperature_breach('PASSIVE_COOLING', None)== "Wrong Input")
+    self.assertTrue(typewise_alert.classify_temperature_breach(None, None)== "Wrong Input")
+       
+  def test_check_and_alert(self):
+    self.assertTrue(typewise_alert.check_and_alert('TO_CONSOLE',{'coolingType':'HIGH_ACTIVE_COOLING'},100)== True)
```

Comment:
These asserts state that `check_and_alert` converts the given parameters to boolean. If you had to state its function as a data-conversion activity, what would that be?

Reply:
Hi Sudeep,
I didnt get your comment can you please elaborate a bit. thanks

Reply:
Your tests will pass even if check_and_alert sends email instead of sending to controller (let's say there is a mistake in the `alert_Target_Type` dictionary). Or if the to-field is left out in the email.
That means even after your tests pass, you will need to check your code by integrating with real controllers, emailers, etc.
To strengthen your tests, make a series of data-transformations and test them with return values. Only the dependency part will be left then.